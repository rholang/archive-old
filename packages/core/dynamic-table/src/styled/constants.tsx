import { css } from 'styled-components';
import { gridSize } from '@atlaskit/theme/constants';
import { N30A } from '@atlaskit/theme/colors';
import { divide } from '@atlaskit/theme/math';
import { ASC, DESC } from '../internal/constants';
import { arrow } from '../theme';
import { SortOrderType } from '../types';

export interface TruncateStyleProps {
  width?: number;
  isFixedSize?: boolean;
  shouldTruncate?: boolean;
}

export const truncateStyle = ({
  width,
  isFixedSize,
  shouldTruncate,
}: TruncateStyleProps) => css`
  ${width
    ? css`
        width: ${width}%;
      `
    : ''} ${isFixedSize
    ? css`
        overflow: hidden;
      `
    : ''};
  ${isFixedSize && shouldTruncate
    ? css`
        text-overflow: ellipsis;
        white-space: nowrap;
      `
    : ''};
`;

export const onClickStyle = ({ onClick }: { onClick?: boolean }) =>
  onClick &&
  css`
    &:hover {
      cursor: pointer;
      background-color: ${N30A};
    }
  `;

export const arrowsStyle = (props: {
  isSortable?: boolean;
  sortOrder?: SortOrderType;
}) => {
  const { isSortable, sortOrder } = props;

  if (!isSortable) {
    return '';
  }

  const pseudoBase = css`
    border: 3px solid transparent;
    display: block;
    height: 0;
    position: absolute;
    right: -${gridSize}px;
    width: 0;
  `;

  return css`
    & > span {
      position: relative;
      &::before {
        ${pseudoBase};
        border-bottom: 3px solid
          ${sortOrder === ASC
            ? arrow.selectedColor(props)
            : arrow.defaultColor(props)};
        bottom: 8px;
        content: ' ';
      }
      &::after {
        ${pseudoBase};
        border-top: 3px solid
          ${sortOrder === DESC
            ? arrow.selectedColor(props)
            : arrow.defaultColor(props)};
        bottom: 0;
        content: ' ';
      }
    }

    &:hover > span {
      &::before {
        border-bottom: 3px solid
          ${sortOrder === ASC
            ? arrow.selectedColor(props)
            : arrow.hoverColor(props)};
      }
      &::after {
        border-top: 3px solid
          ${sortOrder === DESC
            ? arrow.selectedColor(props)
            : arrow.hoverColor(props)};
      }
    }
  `;
};

export const cellStyle = css`
  border: none;
  padding: ${divide(gridSize, 2)}px ${gridSize}px;
  text-align: left;

  &:first-child {
    padding-left: 0;
  }
  &:last-child {
    padding-right: 0;
  }
`;
