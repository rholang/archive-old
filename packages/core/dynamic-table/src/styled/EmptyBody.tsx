import styled from 'styled-components';
import { gridSize } from '@atlaskit/theme/constants';
import { multiply } from '@atlaskit/theme/math';

export const EmptyViewWithFixedHeight = styled.div`
  height: ${multiply(gridSize, 18)}px;
`;

export const EmptyViewContainer = styled.div`
  margin: auto;
  padding: 10px;
  text-align: center;
  width: 50%;
`;
