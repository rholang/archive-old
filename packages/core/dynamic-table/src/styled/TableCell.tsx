import styled from 'styled-components';
import { TruncateStyleProps, truncateStyle, cellStyle } from './constants';

export const TableBodyCell = styled.td<TruncateStyleProps>`
  ${props => truncateStyle(props)} ${cellStyle};
`;
