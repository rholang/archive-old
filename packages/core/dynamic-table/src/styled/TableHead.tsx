import styled, { css } from 'styled-components';
import { B100 } from '@atlaskit/theme/colors';
import {
  arrowsStyle,
  cellStyle,
  onClickStyle,
  truncateStyle,
  TruncateStyleProps,
} from './constants';
import { SortOrderType } from '../types';
import { head } from '../theme';

const rankingStyles = css`
  display: block;
`;

interface HeadProps {
  isRanking?: boolean;
}

export const Head = styled.thead`
  border-bottom: 2px solid ${head.borderColor};
  ${({ isRanking }: HeadProps) => isRanking && rankingStyles};
`;

interface HeadCellProps extends TruncateStyleProps {
  onClick?: () => void;
  isSortable?: boolean;
  sortOrder?: SortOrderType;
}

export const HeadCell = styled.th<HeadCellProps>`
  ${p => onClickStyle({ onClick: Boolean(p.onClick) })} ${p =>
  truncateStyle(p)} ${p => arrowsStyle(p)} ${cellStyle} border: none;
  color: ${head.textColor};
  box-sizing: border-box;
  font-size: 12px;
  font-weight: 600;
  position: relative;
  text-align: left;
  vertical-align: top;
  &:focus {
    outline: solid 2px ${B100};
  }
`;
