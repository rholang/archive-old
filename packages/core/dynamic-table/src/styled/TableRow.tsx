import styled, { css } from 'styled-components';
import { row } from '../theme';

export interface ITableRowProps {
  isHighlighted?: boolean;
}

export const TableBodyRow = styled.tr<ITableRowProps>`
  ${({ isHighlighted }) =>
    isHighlighted &&
    css`
      background-color: ${row.highlightedBackground};
    `}

  &:hover {
    ${({ isHighlighted }) =>
      css`
        background-color: ${isHighlighted
          ? row.hoverHighlightedBackground
          : row.hoverBackground};
      `}
`;
