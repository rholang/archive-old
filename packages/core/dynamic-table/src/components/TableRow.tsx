import React from 'react';
import { TableBodyRow } from '../styled/TableRow';
import { TableBodyCell } from '../styled/TableCell';
import { HeadType, RowType } from '../types';

interface Props {
  head?: HeadType;
  isFixedSize: boolean;
  isHighlighted?: boolean;
  row: RowType;
}

const Row = ({ row, head, isFixedSize, isHighlighted }: Props) => {
  const { cells, ...restRowProps } = row;

  return (
    <TableBodyRow {...restRowProps} isHighlighted={isHighlighted}>
      {cells.map((cell, cellIndex) => {
        const { content, ...restCellProps } = cell;
        const { shouldTruncate, width } =
          (head || { cells: [] }).cells[cellIndex] || ({} as any);

        return (
          <TableBodyCell
            {...restCellProps}
            isFixedSize={isFixedSize}
            key={cellIndex} // eslint-disable-line react/no-array-index-key
            shouldTruncate={shouldTruncate}
            width={width}
          >
            {content}
          </TableBodyCell>
        );
      })}
    </TableBodyRow>
  );
};

export default Row;
