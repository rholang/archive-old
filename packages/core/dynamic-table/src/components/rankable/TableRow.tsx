import React from 'react';
import { Draggable } from 'react-beautiful-dnd';
import { RankableTableBodyRow } from '../../styled/rankable/TableRow';
import { HeadType, RowType } from '../../types';
import withDimensions, { WithDimensionsProps } from '../../hoc/withDimensions';
import TableCell from './TableCell';
import { inlineStylesIfRanking } from '../../internal/helpers';

export interface Props extends WithDimensionsProps {
  head?: HeadType;
  isFixedSize: boolean;
  row: RowType;
  rowIndex: number;
  isRankingDisabled: boolean;
  isHighlighted?: boolean;
}

export class RankableTableRow extends React.Component<Props, {}> {
  innerRef = (innerRefFn: Function) => (ref?: HTMLElement) => {
    innerRefFn(ref);
    if (this.props.innerRef) {
      this.props.innerRef(ref);
    }
  };

  render() {
    const {
      row,
      head,
      isFixedSize,
      isRanking,
      refWidth,
      rowIndex,
      isRankingDisabled,
      isHighlighted,
    } = this.props;
    const { cells, key, ...restRowProps } = row;
    const inlineStyles = inlineStylesIfRanking(isRanking, refWidth);

    if (typeof key !== 'string' && !isRankingDisabled) {
      throw new Error(
        'dynamic-table: ranking is not possible because table row does not have a key. Add the key to the row or disable ranking.',
      );
    }

    return (
      <Draggable
        draggableId={key || `${rowIndex}`}
        index={rowIndex}
        isDragDisabled={isRankingDisabled}
      >
        {(provided, snapshot) => (
          <RankableTableBodyRow
            {...restRowProps}
            {...provided.dragHandleProps}
            {...provided.draggableProps}
            innerRef={this.innerRef(provided.innerRef)}
            style={{ ...provided.draggableProps.style, ...inlineStyles }}
            isHighlighted={isHighlighted}
            isRanking={isRanking}
            isRankingItem={snapshot.isDragging}
          >
            {cells.map((cell, cellIndex) => {
              const headCell = (head || { cells: [] }).cells[cellIndex];
              return (
                <TableCell
                  head={headCell}
                  cell={cell}
                  isRanking={isRanking}
                  key={cellIndex} // eslint-disable-line react/no-array-index-key
                  isFixedSize={isFixedSize}
                />
              );
            })}
          </RankableTableBodyRow>
        )}
      </Draggable>
    );
  }
}

export default withDimensions<Props>(RankableTableRow);
