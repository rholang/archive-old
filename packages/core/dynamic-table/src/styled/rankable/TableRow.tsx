import styled, { css } from 'styled-components';
import { N20, B100 } from '@atlaskit/theme/colors';
import { e500 } from '@atlaskit/theme/elevation';
import { TableBodyRow } from '../TableRow';

const rankingStyles = css`
  display: block;
`;

/**
 * TODO: Pass the props here to get particular theme for the table
 * Skipping it for now as it may impact migration as util-shared-styles does not support this feature
 */
const rankingItemStyles = css`
  background-color: ${N20};
  ${e500()} border-radius: 2px;
`;

const draggableStyles = ({
  isRanking,
  isRankingItem,
}: {
  isRanking?: boolean;
  isRankingItem?: boolean;
}) => css`
  ${isRanking && rankingStyles} ${isRankingItem && rankingItemStyles} &:focus {
    outline-style: solid;
    outline-color: ${B100};
    outline-width: 2px;
  }
`;

export const RankableTableBodyRow = styled(TableBodyRow)`
  ${draggableStyles};
`;
