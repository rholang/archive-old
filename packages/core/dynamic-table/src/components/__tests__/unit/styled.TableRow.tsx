import React from 'react';
import { TableBodyRow } from '../../../styled/TableRow';
import { shallow, ShallowWrapper } from 'enzyme';

describe('TableRow', () => {
  let shallowWrapper: ShallowWrapper;

  describe('on rendering with highlighted prop', () => {
    beforeEach(() => {
      shallowWrapper = shallow(<TableBodyRow isHighlighted={true} />);
    });

    it('should render with required background color', () => {
      expect(shallowWrapper).toMatchSnapshot();
    });
  });

  describe('on rendering without highlighted prop', () => {
    beforeEach(() => {
      shallowWrapper = shallow(<TableBodyRow isHighlighted={false} />);
    });

    it('should render without background color', () => {
      expect(shallowWrapper).toMatchSnapshot();
    });
  });
});
