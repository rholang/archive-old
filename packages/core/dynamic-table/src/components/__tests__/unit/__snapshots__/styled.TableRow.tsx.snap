// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow on rendering with highlighted prop should render with required background color 1`] = `
.c0 {
  background-color: #F4F5F7;
}

.c0:hover {
  background-color: #EBECF0;
}

<tr
  className="c0"
/>
`;

exports[`TableRow on rendering without highlighted prop should render without background color 1`] = `
.c0:hover {
  background-color: #FAFBFC;
}

<tr
  className="c0"
/>
`;
